### ISSUE TITLE
Allow opening file instead of diff

### ISSUE THREAD

**Issue by rryk** [eamodio/vscode-gitlens#1651]
When clicking on the filename in "Search & Compare", it opens a diff by default. To open the file itself one needs to right-click and select "Open File". Would be nice to have a setting to open files by default. This is similar to https://github.com/microsoft/vscode/issues/36885, but for GitLens panels.

**Comment by gspencergoog** [comment]
This is the default behavior for Gitscope on IntellliJ, and I find it to be incredibly helpful there.  It lets me use the current list of diffs from the main branch as a workspace that contains only the files I'm working on for the current change. With the need to click on a small icon to open the file instead of the diff, it makes me think about opening a file more carefully, and slows me down.

**Comment by eamodio** [comment]
Sounds like a good enhancement. Thanks!

**Comment by gspencergoog** [comment]
The VSCode git extension has an option called "git.openDiffOnClick" that does exactly what this feature request describes. It would be nice if GitLens had this too.

**Comment by marcoapcaldas** [comment]
Is this option inline sufficient for this feature?

![image](https://user-images.githubusercontent.com/37601361/218082689-6fefe49d-1aba-420f-9348-cb1abfc63113.png)

**Comment by gspencergoog** [comment]
@marcoapcaldas No, because it's about the default: both the diff and the file icons are available, but the diff is the default when you click on it, and I'd much prefer it be the file. I'm constantly accidentally opening the diff window, immediately closing it, and then having to focus to click the tiny file icon, which takes me out of my flow.

**Comment by marcoapcaldas** [comment]
I'm trying to find which files I should work on to make this feature, but I hadn't sucess. Could you give me some tips @eamodio ?

**Comment by EhabY** [comment]
Given that the old PR is more than a year old, I have submitted another PR to fix this issue https://github.com/gitkraken/vscode-gitlens/pull/3583

**Comment by EhabY** [comment]
Resolved the conflicts and updated my solution, please take a look if this is still wanted 🙏

**Comment by DanFessler** [comment]
Here to voice my support for this feature.
